core/dbt/contracts/graph/manifest.py

```diff
@@ -936,6 +938,31 @@ class Manifest(MacroMethods, dbtClassMixin):
         metadata={"serialize": lambda x: None, "deserialize": lambda x: None},
     )
 
+    # This and the following methods can be refacong into a mixin
+    # and we should do so next time we do it for another artifact.
+    @classmethod
+    def validate(cls, data: Any) -> None:
+        json_schema = cls.json_schema()
+        cls._allow_additional_recursive(cls, json_schema)
+        validator = jsonschema.Draft7Validator(json_schema)
+        error = next(iter(validator.iter_errors(data)), None)
+        if error is not None:
+            raise ValidationError.create_from(error) from error
+
+    @staticmethod
+    def _allow_additional_recursive(cls, dct: Dict[str, Any]):
```

Comment:
I prefer a smaller interface to start with and if we need to make it generic it is pretty easy and we know more what it should look like at the time